### PR TITLE
tooling: Use `envoy_entry_point`

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:

This is a follow up to use the macro added in #19450 for existing entry_points

Also contains a fix to use a template to create the py_binary main to ensure the py_binary retains the path to the entry point

Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/base/entry_point.py [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
 import subprocess
 import sys
 
+ENTRY_POINT_ALIAS = "_ENTRY_POINT_ALIAS_"
+
+
+def find_tool_path():
+    entry_point_alias = ENTRY_POINT_ALIAS.split("/external/")[1]
+
+    for x in pathlib.Path(".").glob(f"**/{entry_point_alias}"):
```
Comment (htuch) [inline tools/base/entry_point.py:11]: This seems a bit of a wide net to cast. PR LGTM otherwise.

Comment (phlax) [reply tools/base/entry_point.py:11]: the entry point alias that its matching is a pretty long string - if it matches we can be fairly sure we found the binary

i couldnt find any other way to do it - not sure why but the location im getting back from the alias wasnt matching exactly,  even tho the tool is actually there (on a slightly different filepath)

Comment (htuch) [reply tools/base/entry_point.py:11]: Can you provide some concrete path examples? I think it's probably OK as a workaround, but it just has a code smell to it, where we might end up with some hard to diagnose bug later due to this use of globbing.

Comment (phlax) [reply tools/base/entry_point.py:11]: >  it just has a code smell to it

yep, agreed, i spent quite a few hours on this, and didnt find an alternative - so reluctantly did it this way - ill dump what is being mangled now and add as a comment...

Comment (phlax) [reply tools/base/entry_point.py:11]: hmm, not sure about adding a comment as the lines are very long, ill post here for now and we can figure out

essentially there are 2 situations that it needs to work in - when its called directly with `bazel run //path/to:entry_point` and where are a build rule calls it - ie `bazel build //rule/that/calls:entry_point`

This is the before/after mangling for the first - ie bazel run

```console
ENTRY_POINT_ALIAS: bazel-out/k8-opt-exec-2B5CBBC6/bin/external/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner
FOUND PATH: external/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner

```

And this is for the second:

```console

ENTRY_POINT_ALIAS: bazel-out/host/bin/external/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner
FOUND PATH: bazel-out/host/bin/tools/docs/sphinx_runner.runfiles/base_pip3_envoy_docs_sphinx_runner/rules_python_wheel_entry_point_envoy.docs.sphinx_runner

```

Comment (phlax) [reply tools/base/entry_point.py:12]: ive cleaned the macro up a little, but whatever way i look at it the path that the genrule sees is different from the path that the py_binary sees 

im pretty confident that this workaround works, not sure how else to achieve the desired outcome

Comment (htuch) [reply tools/base/entry_point.py:12]: OK. The only other suggestions I would have here are to think of (1) passing the path in via command-line from Bazel which should know what it is or (2) using Bazel query. @lizan might also have some insight. In any case, it's probably not worth spending a ton of time optimizing, so maybe just document why this is the way it is and we can land.

Comment (phlax) [reply tools/base/entry_point.py:12]: passing via the path wont work - its the reason for shifting to using a template - the arg gets lost when the rule is called from another rule, ill add a comment now...

**Review by htuch — APPROVED** [review]
Ack, thanks, I'm going to chalk this one up to another Bazel workaround, which we have plenty of, thanks.

**Pull request merged**